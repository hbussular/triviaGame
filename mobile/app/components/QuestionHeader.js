import React from 'react'
import PropTypes from 'prop-types'
import { Text, View } from 'react-native'

import styles from './QuestionHeader.style'

const QuestionHeader = ({
  questionPosition,
  questionsLength,
}) => (
  <View>
    <Text style={styles.textContainer}>
      {`Question ${questionPosition} of ${questionsLength}`}
    </Text>
  </View>
)

QuestionHeader.propTypes = {
  questionPosition: PropTypes.number.isRequired,
  questionsLength: PropTypes.number.isRequired,
}

export default QuestionHeader
