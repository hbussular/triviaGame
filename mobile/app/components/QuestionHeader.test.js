import React from 'react'
import renderer from 'react-test-renderer'
import QuestionHeader from './QuestionHeader'

test('renders correctly', () => {
  const props = { questionsLength: 10, questionPosition: 1 }
  const tree = renderer
    .create(<QuestionHeader {...props} />)
    .toJSON()

  expect(tree).toMatchSnapshot()
})
