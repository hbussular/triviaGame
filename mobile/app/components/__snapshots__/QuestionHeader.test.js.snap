// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "rgba(255, 255, 255, 1)",
        "fontSize": 38,
        "fontWeight": "700",
        "padding": 20,
        "textAlign": "center",
      }
    }
  >
    Question 1 of 10
  </Text>
</View>
`;
